### PR TITLE
Added geometry tests

### PR DESCRIPTION
**Pull request by walkerar**
Maybe some tests?

### CODE REVIEW

**Comment by coltonbh** [comment]
@walkerar Note the failing `flake8` and `black` checks here on this PR. You'll want to flake8 and blacken your code so the formatting is good. Click on "details" above to see where the failures are. You can see which files need fixing!

You can run `black path/to/file.py` to have black format the file for you.
And you can run `flake8 path/to/file.py` to see what code formatting errors need fixing :)

**Comment by walkerar** [comment]
I think I fixed it? I got a cake icon :D

**Comment by coltonbh** [comment]
/

> I think I fixed it? I got a cake icon :D

YEAH! Easy and pie and sweet as cake to have clean, formatted code 🎂✨

Now start using a real code editor and it can do it for you automagically every time you save ;)

File: tests/test_geom_operations.py
Comment (coltonbh) [inline tests/test_geom_operations.py:10]: You don't use `tmp_path` in your function so it is not needed :)

This is a way to ask `pytest` to pass you a temporary directory you can use on a machine for saving scratch files to.

File: tests/test_geom_operations.py
Comment (coltonbh) [inline tests/test_geom_operations.py:20]: Note what I've done here. I've create a `trajectory` test fixture. This means your test can just "ask" for the fixture in your tests and `pytest` will deliver it to them. You can see how I created this fixture in the `conftest.py` file, if interested.

I've also refactored our one of your tests so that `test_compute_bond` stands alone as its own unit tests. Your time to practice now! See comment on `test_geom_operations`.

File: tests/test_geom_operations.py
Comment (coltonbh) [inline tests/test_geom_operations.py:35]: Please refactor this tests into two separate tests, `def test_compute_angle` and `def test_compute_torsion` following the pattern that I gave you on `def test_compute_bond`. 

Note how now each tests properly tests only one piece of functionality (a good unit test!) and no longer has to contain code that is irrelevant to the unit under test (no longer have to create traj objects, merge them together, etc...). Any time we see something like that that is repeated, it's a good sign it belongs somewhere else and a refactor is warranted.

**Review by coltonbh — CHANGES_REQUESTED** [review]
Thank you so much for adding these tests and for the note that steps 1,2,3 of your previous tests need to be repeated for each test (that's very helpful for me to know what should be refactored into a test fixture :) )

Please refactor your test into separate tests, as per my comment below, that leverage the test fixture by listing it in their function signature and ensuring each tests tests only one unit of coded.

After that I'll squash and merge code. Great work 😀

File: tests/test_geom_operations.py
Comment (coltonbh) [inline tests/test_geom_operations.py:18]: What exactly is this testing? We are testing that the number of frames before/after the computation is the same. Is that a critical value we'd expect to change, indicating that something went wrong?

Comment (walkerar) [reply tests/test_geom_operations.py:18]: Removed as per slack convo :)

File: tests/test_geom_operations.py
Comment (walkerar) [reply tests/test_geom_operations.py]: Refactored angle/torsion tests

**Comment by coltonbh** [comment]
Ok @walkerar note the failing code quality checks (isort and flake8). You can fix the offending files with `isort` just like using `black` by running `isort path/to/file.py`. 

Have you addressed the comments above? If so click on "Resolve conversation" so that we know those issues are taken care of and then just make a comment here with "Hey @coltonbh mind taking another look?" and then I know to move the process forward :)

File: tests/test_geom_operations.py
Comment (walkerar) [reply tests/test_geom_operations.py]: Replaced with the generic trajectory pass

File: tests/test_geom_operations.py
Comment (walkerar) [reply tests/test_geom_operations.py]: Redid the tests to be broken up and to use the trajectory test fixture

File: tests/test_geom_operations.py
Comment (walkerar) [reply tests/test_geom_operations.py]: temp_path has been removed

**Comment by walkerar** [comment]
@coltonbh I believe I have fixed most of the issues, though I am unsure how to fix the documentation checks.

**Comment by coltonbh** [comment]
@walkerar looks good! I made one change to the fixture to make it run faster. Ready to merge this in or are there addition tests you anticipated adding with this PR?

**Comment by walkerar** [comment]
I added all the geom checks, more tests will be in other files and stuff so I think we should merge :)

**Comment by coltonbh** [comment]
Ok lemme squash to make it all pretty. Nice work!

**Comment by coltonbh** [comment]
Hold tight and you can smash that merge button in just a second...

**Review by coltonbh — APPROVED** [review]
Looks good! Hit that merge button, then delete the extra message they offer by default. Then click merge!

**Pull request merged**
